linalg/hypre.cpp

```diff
@@ -1573,9 +1573,16 @@ HypreParMatrix *HypreParMatrix::ExtractSubmatrix(const Array<int> &indices,
    }
 
    // Construct cpts_global array on hypre matrix structure
+#if defined(HYPRE_DEVELOP_BRANCH) && HYPRE_DEVELOP_NUMBER >= 8
```

Comment:
Shouldn't there be a check here about the hypre version i.e. state that we are at least 8 commits after `v2.23.0`?

Reply:
Hi @tzanio, I assumed that `HYPRE_DEVELOP_BRANCH` refers to the current `hypre-master`, which is ahead of `v2.23.0`. However, this would need to be updated to
```
#if MFEM_HYPRE_VERSION > 22300
```
after the next release of hypre. Alternatively, I though about having:
```
#if (MFEM_HYPRE_VERSION > 22300) || 
    (MFEM_HYPRE_VERSION == 22300 && defined(HYPRE_DEVELOP_BRANCH) && HYPRE_DEVELOP_NUMBER >= 8)
```
which would work for the current hypre-master and next releases as well. Does the last option sounds reasonable?

Reply:
I like the last version, because we don't need to remember to update the code after the next hypre release.

In the future I think we can just say

```
#if (MFEM_HYPRE_VERSION > 22400) || (MFEM_HYPRE_VERSION == 22400 && HYPRE_DEVELOP_NUMBER >= 8)
```

Reply:
Sorry for the delay @tzanio, I updated the code according to your suggestion in ba1e213. Thanks!